tests/test_masterclasses.py

```diff
@@ -164,18 +177,38 @@ def test_draft_masterclass_id_is_added_to_session(logged_in_user, blank_session)
     with logged_in_user.post('/create-masterclass'):
         assert session['draft_masterclass_id'] == 1
 
-def test_add_existing_content_to_draft_masterclass(logged_in_user, db, new_masterclass_content_data_category, new_masterclass, blank_session):
+def test_add_existing_content_to_draft_masterclass(logged_in_user, db, new_masterclass_content_data_category, new_masterclass_with_location, blank_session):
     with logged_in_user.session_transaction() as session:    
         session['draft_masterclass_id'] = 1
     masterclass_content = MasterclassContent.query.filter_by(name='Introduction to R').first()
     logged_in_user.post('/create-masterclass/content/new-or-existing', data = {'which-masterclass': masterclass_content.id})
     draft_masterclass = Masterclass.query.get(1)
     assert draft_masterclass.masterclass_content_id == masterclass_content.id
 
-def test_add_new_content_to_draft_masterclass(logged_in_user, new_masterclass, blank_session):
+def test_add_new_content_to_draft_masterclass(logged_in_user, new_masterclass_with_location, blank_session):
     with logged_in_user.session_transaction() as session:    
         session['draft_masterclass_id'] = 1
     logged_in_user.post('/create-masterclass/content/create-new', data = {'masterclass-name': 'Test name', 'masterclass-description': 'A description'})
     draft_masterclass = Masterclass.query.get(1)
     new_content = MasterclassContent.query.filter_by(name='Test name').first()
     assert draft_masterclass.masterclass_content_id == new_content.id
+
+@pytest.mark.parametrize('location_details_to_display, masterclass_id',
+(
+('Test building', 1),
+('Remote', 2)
+))
+def test_correct_location_details_displayed(logged_in_user, new_masterclass_remote, new_masterclass_with_location, blank_session, location_details_to_display, masterclass_id, new_location):
+    response = logged_in_user.get(f'/masterclass/{masterclass_id}')
+    assert f'<p class="govuk-body">{location_details_to_display}</p>' in response.get_data(as_text=True)
+
+def test_remote_joining_info_hidden_in_masterclass_profile_if_user_not_attendee(logged_in_user, new_masterclass_remote, blank_session):
+    response = logged_in_user.get('/masterclass/2')
+    assert '<h2 class="govuk-heading-m">Joining link</h2>' not in response.get_data(as_text=True)
+
+def test_remote_joining_info_visible_in_masterclass_profile_if_user_is_attendee(logged_in_user, new_masterclass_remote, blank_session):
+    # Sign current user up to masterclass
+    logged_in_user.post('/masterclass/2')
+    response = logged_in_user.get('/masterclass/2')
+    assert '<h2 class="govuk-heading-m">Joining link</h2>' in response.get_data(as_text=True)
+    # This not working
```

Comment:
Are you committing a non-working test?

Reply:
Ah I had forgotten to take that out once I got it working again. Have now removed.